lib/network_utils.pm

```diff
@@ -137,4 +137,22 @@ sub ifc_is_up {
     return !script_run("ip link show dev $ifc | grep 'state UP'");
 }
 
+=head2 genmac
+
+Generate custom MAC address.
+Used for Xen domU testing, to define MAC address once for whole test suite lifecycle.
+
+ genmac(['aa:bb:cc'])
+
+=cut
+
+sub genmac {
+    my @mac = split(/:/, shift);
+    my $len = scalar(@mac);
+    for (my $i = 0; $i < (6 - $len); $i++) {
+        push @mac, (sprintf("%02X", int(rand(255))));
```

Comment:
On the second though this can generate one pair as `FF`. Not sure, whether it can be a problem in some corner case or not.

Reply:
```suggestion
        push @mac, (sprintf("%02X", int(rand(254))));
```

Perhaps this solved the issue already. This excludes `FF` while still giving enough space for for randomness.